apis/src/api/v1/bot/games.rs

```diff
@@ -0,0 +1,63 @@
+use crate::api::v1::auth::auth::Auth;
+use actix_web::get;
+use actix_web::web::{Data, Json};
+use actix_web::{post, web, HttpResponse};
+use anyhow::{anyhow, Result};
+use db_lib::{
+    get_conn,
+    models::{Game, User},
+    DbPool,
+};
+use diesel_async::scoped_futures::ScopedFutureExt;
+use diesel_async::AsyncConnection;
+use hive_lib::State;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use shared_types::GameId;
+
+#[derive(Serialize, Deserialize)]
+struct GameRequest {
+    selector: GameSelector,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum GameSelector {
+    Ongoing,
+    Pending,
+    Specific(GameId),
+}
+
+#[post("/api/v1/bot/games")]
+pub async fn api_get_games(
+    Json(req): Json<GameRequest>,
+    Auth(email): Auth,
+    pool: Data<DbPool>,
+) -> HttpResponse {
+    match get_games(&email, req.selector, pool).await {
+        Ok(games) => HttpResponse::Ok().json(json!({
+          "success": true,
+          "data": {
+            "user": email,
+            "games": games,
```

Comment:
So, if I understand correctly, the whole game object is going to be serialized ? If I look at `models::Game` I see that we have the players UUIDs but not their name. Shouldn't we give that information ?

Reply:
Currently yes, but that's just because I was a) lazy b) didn't know which fields we would like to keep/add. Once we have that I am going to write a serializer that just send all the necessary info. But be careful: usernames can change. So using UUID might be the most reliable thing. I can write you a small uuid -> user (profile) api endpoint.

Reply:
And endpoint to translate uuid to user would be interesting. Especially if you give more information about the user that way: like their elo rating